core/src/main/java/org/elasticsearch/common/settings/SecureString.java

```diff
@@ -182,14 +180,15 @@ private void ensureNotClosed() {
     @Override
     public synchronized void writeTo(StreamOutput out) throws IOException {
         ensureNotClosed();
-        ByteBuffer byteBuffer = null;
+        byte[] bytes = null;
         try {
-            byteBuffer = StandardCharsets.UTF_8.encode(CharBuffer.wrap(chars));
-            out.writeVInt(byteBuffer.limit());
-            out.writeBytes(byteBuffer.array(), byteBuffer.arrayOffset(), byteBuffer.limit());
+            final int utf8Size = UnicodeUtil.calcUTF16toUTF8Length(new CharsRef(chars, 0, chars.length), 0, chars.length);
+            bytes = new byte[utf8Size];
```

Comment:
move this out of the try block  and make the bytes final? then you can also skip the null check below